### ISSUE TITLE
confuse about the error return  in grpc call when ctx is Done

### ISSUE THREAD

**Issue by lqczzz** [micro/go-micro#720]
https://github.com/micro/go-micro/blob/956be5c59d28d2f437eb383783091a7604013540/client/grpc/grpc.go#L138

why doesn't the err from ctx wrapped by micro error?

i suppose i should look like below?
``` go
	select {
	case err := <-ch:
		grr = err
	case <-ctx.Done():
		grr = microError(ctx.Err())
	}
```

**Comment by qiluoao** [comment]
shouldn't it be like:
```go
errors.New("go.micro.client", fmt.Sprintf("%v", ctx.Err()), 408)
```

**Comment by milosgajdos** [comment]
Can you open a PR @lqczzz ?

**Comment by vtolstov** [comment]
fixed by #1166
